src/plot_api/helpers.js

```diff
@@ -439,11 +439,14 @@ function commonPrefix(name1, name2, show1, show2) {
 function cleanTextPosition(textposition) {
     var posY = 'middle',
         posX = 'center';
-    if(textposition.indexOf('top') !== -1) posY = 'top';
-    else if(textposition.indexOf('bottom') !== -1) posY = 'bottom';
 
-    if(textposition.indexOf('left') !== -1) posX = 'left';
-    else if(textposition.indexOf('right') !== -1) posX = 'right';
+    if(textposition !== undefined && textposition !== null) {
```

Comment:
I think we have to be even more strict at this stage here. Anything that's not a string shouldn't fall into this block. The items inside `textposition` can really be anything.

Have you tried something like:

```
Plotly.newPlot(gd, [{
  type: 'scatter3d',
  mode: 'markers+text',
  x: [1, 2],
  y: [1, 2],
  z: [1, 2],
  text: ['a', 'b'],
  textposition: [null, undefined, true, false, [], {}, NaN, Infinity]
}])
```

Reply:
... and would you mind adding a jasmine test for this :arrow_heading_up: 

Somewhere at the end of this [describe block](https://github.com/plotly/plotly.js/blob/32cdbbea8916cea45a7c2389482346e1555a5099/test/jasmine/tests/gl3d_plot_interact_test.js#L24) would be :ok_hand: